modules/compute/gke-node-pool/main.tf

```diff
@@ -70,7 +70,10 @@ data "google_container_cluster" "gke_cluster" {
 resource "google_container_node_pool" "node_pool" {
   provider = google-beta
 
-  name           = coalesce(var.name, "${var.machine_type}-${local.module_unique_id}")
+  count = var.num_node_pools
+
+  # name           = var.num_node_pools == 1 ? coalesce(var.name, var.machine_type+"-"+local.module_unique_id) : coalesce(var.name, var.machine_type+"-"+local.module_unique_id)+"-"+count.index
+  name           = var.num_node_pools == 1 ? coalesce(var.name, join("-", [var.machine_type, local.module_unique_id])) : join("-", [coalesce(var.name, join("-", [var.machine_type, local.module_unique_id])), count.index])
```

Comment:
what is the result of this name pattern? I thought the id can be a number, will it just append the numbers?  e.g. foo-tpuv4-08751 vs foo-tpuv4-0875 ? should you add another dash before the index?

Reply:
There is already a "-" that joins the name to the count.index
So, for the example you mentioned, it will be `foo-tpuv4-0875-1`
Is this the expected result of your idea?

Reply:
that works, thanks!